cmd/containerboot/main.go

```diff
@@ -559,25 +559,26 @@ runLoop:
 					log.Println("Startup complete, waiting for shutdown signal")
 					startupTasksDone = true
 
-					// Reap all processes, since we are PID1 and need to collect zombies. We can
-					// only start doing this once we've stopped shelling out to things
-					// `tailscale up`, otherwise this goroutine can reap the CLI subprocesses
-					// and wedge bringup.
+					// Wait on tailscaled process. It won't
+					// be cleaned up by default when the
+					// container exits as it is not PID1.
+					// TODO (irbekrm): perhaps we can
+					// replace the reaper by a running
+					// cmd.Wait in a goroutine immediately
+					// after starting tailscaled?
```

Comment:
We could probably remove the reaper altogether and just run `cmd.Wait` [here-ish](https://github.com/tailscale/tailscale/blob/v1.64.2/cmd/containerboot/main.go#L629) in a goroutine. But I am not sure if that would be much clearer to read, is not functionally much different and I am reluctant to change too many things at once here.

Reply:
This is fine, but yeah definitely using cmd.Wait is a lot safer and simpler

Reply:
I'll merge this as is now to fix the bug and look at a bigger cleanup later